FEAT: 버그제보 기록 API #11

bsh714-dev opened the pull request:
---
name: Pull Request
about: PR을 위해 작업한 상세 내용 작성
title: '[커밋 컨벤션 헤더] 제목'
labels: ''
assignees: ''

---

**작업한 내용**
버그제보 기록 API


**REVIEW POINT**
_이번 PR에서 리뷰해야 할 포인트가 있다면 적어주세요._

bsh714-dev committed:
FEAT: socketIdMap에 존재하는 회원인지 검증하는 authGuard 개발 #43

On branch feat/43
 Changes to be committed:
	new file:   dist/src/logger.service.d.ts
	new file:   dist/src/logger.service.js
	new file:   dist/src/logger.service.js.map
	new file:   dist/src/middlewares/auth-guard.middleware.d.ts
	new file:   dist/src/middlewares/auth-guard.middleware.js
	new file:   dist/src/middlewares/auth-guard.middleware.js.map
	modified:   dist/src/middlewares/logger.middleware.d.ts
	modified:   dist/src/middlewares/logger.middleware.js
	modified:   dist/src/middlewares/logger.middleware.js.map
	modified:   dist/tsconfig.build.tsbuildinfo
	modified:   package-lock.json
	modified:   package.json
	new file:   src/logger.service.ts
	new file:   src/middlewares/auth-guard.middleware.ts
	modified:   src/middlewares/logger.middleware.ts

bsh714-dev committed:
logging

bsh714-dev committed:
pra2

bsh714-dev committed:
Merge branch 'feat/43' of https://github.com/gachimind/backend into prack1

bsh714-dev committed:
Merge branch 'dev' of https://github.com/gachimind/backend into feature/134

bsh714-dev committed:
FEAT : 노션 API를 이용한 게임데이터 수집기능 개발 #134

bsh714-dev committed:
FEAT : 노션 API를 이용한 게임데이터 수집기능 개발 #134

On branch feat/134

bsh714-dev committed:
conflict 제거

bsh714-dev committed:
Merge branch 'dev' of https://github.com/gachimind/backend into feat/134

bsh714-dev committed:
ff

bsh714-dev committed:
FEAT: 버그제보 기록 API #11

bsh714-dev committed:
feat11

bsh714-dev committed:
모듈삭제

jn33-dev commented on project/src/middlewares/logger.middleware.ts:
```diff
@@ -1,21 +1,28 @@
 @Injectable()
 export class LoggerMiddleware implements NestMiddleware {
-    private logger = new Logger('HTTP');
+    constructor() {}
```
constructor는 사용 안해서 삭제하셔도 될 것 같습니다.

jn33-dev commented on project/src/middlewares/logger.middleware.ts:
```diff
@@ -1,21 +1,28 @@
+                query: _query,
+                body: _body,
+            }),
+        );
```
콘솔에 찍히는 모든 요청에 대한 로깅은 `${method} ${originalUrl} ${statusCode} ${contentLength} - ${userAgent} ${ip}` 이렇게 구성되도록 수정해주세요.
지금은 body, heade query 값이 모두 들어와 굉장히 읽기 힘든것 같아요ㅠ

jn33-dev commented on project/src/logger.service.ts:
```diff
@@ -0,0 +1,77 @@
+                        }),
+                    ),
+                }),
+                new winston.transports.Console({
```
말씀드린대로, 여기 블럭이 콘솔을 찍어주는 역할을 하는 블럭인지 체크해주세요.

jn33-dev commented on project/src/logger.service.ts:
```diff
@@ -0,0 +1,77 @@
+                    ),
+                }),
+                new winston.transports.Console({
+                    level: 'debug',
```
레베 debug로 되어 있는데, 디버그 레벨 로깅이 어떤 걸 포함하고 있는지 확인 부탁드립니다.

jn33-dev commented on project/src/logger.service.ts:
```diff
@@ -0,0 +1,77 @@
+export class LoggerService implements LS {
+    private logger: winston.Logger;
+
+    constructor(service: string) {
```
주입하신 service 체크 부탁드립니다.

jn33-dev commented on project/src/logger.service.ts:
```diff
@@ -0,0 +1,77 @@
+                    level: 'debug',
+                    format: combine(
+                        timestamp({ format: 'isoDateTime' }),
+                        nestWinstonModuleUtilities.format.nestLike(service, {
```
service 주입 수정 후에 콘솔에 찍히는 로깅 포맷이 잘 반영이 된다면, 윈스턴 로거만 두면 될 것 같습니다.

jn33-dev commented on project/src/logger.service.ts:
```diff
@@ -0,0 +1,77 @@
+    }
+    verbose(message: string) {
+        this.logger.verbose(message);
+    }
```
이 레벨링 자동으로 되는지도 다시 한번 확인해주세요!

jn33-dev reviewed (changes requested):
세현님, logger.service 파일 middlewares 폴더로 이동해주세요.

jn33-dev commented on project/src/admin/notion.service.ts:
```diff
@@ -0,0 +1,206 @@
+                        },
+                    ],
+                },
+                일시: {
```
![image](https://user-images.githubusercontent.com/102890891/215790558-54faddd7-477b-49d9-b16c-5e5999c0ddcb.png)

이쪽에서 발생하는 에러 같은데, 이것도 확인 부탁드립니다!

bsh714-dev committed:
FIX:윈스턴 로거 수정

bsh714-dev committed:
FIX : 로거

jn33-dev commented on project/src/admin/admin.controller.ts:
```diff
@@ -7,11 +7,11 @@ export class AdminController {
 
     @Post('report')
-    async bugReport(@Body() Date: BugDto) {
+    async bugReport(@Body() Body: BugDto) {
```
`@Body() Body` 여기에서 Body는 변수명을 지정하는 거라, body 소문자로 사용해주세요.

jn33-dev commented on project/src/app.module.ts:
```diff
@@ -49,7 +50,7 @@ import { TodayResult } from './games/entities/todayResult.entity';
             //migrations: [__dirname + '/migrations/*.ts'],
             // 처음 db를 생성할 때만 synchronize:true로 생성하고, 이 후에는 false로 바꿔야 함
-            synchronize: true,
+            synchronize: false,
```
개발 단계에서 db 스키마 수정이 계속 발생하고 있습니다.
우선 개발 단계에서는 true로 놓고, 배포 후에 false로 변경하면 좋을 것 같습니다.

jn33-dev commented on project/src/middlewares/logger.service.ts:
```diff
@@ -0,0 +1,34 @@
+import * as winston from 'winston';
+
+const logDir = __dirname + '/../../logs';
+const dailyOptions = (level: string) => {
```
parameter로 maxFiles: number도 같이 받도록 추가해주세요.

jn33-dev commented on project/src/middlewares/logger.service.ts:
```diff
@@ -0,0 +1,34 @@
+        datePattern: 'YYYY-MM-DD',
+        dirname: logDir + `/${level}`,
+        filename: `%DATE%.${level}.log`,
+        maxFiles: 30,
```
parameter 값으로 받은 maxFiles 옵션이 적용되도록 : 30 지워주세요

jn33-dev commented on project/src/middlewares/logger.service.ts:
```diff
@@ -0,0 +1,34 @@
+export const winstonLogger = WinstonModule.createLogger({
+    transports: [
+        new winston.transports.Console({
+            level: 'silly',
```
콘솔 옵션으로 아래와 같이 코드 수정해주세요.
```
level: process.env.NODE_ENV === 'production' ? 'http' : 'silly',
    format:
        process.env.NODE_ENV === 'production'
            ? combine(colorize({ all: true }), simple())
            : combine(
                  colorize({ all: true }),
                  timestamp({ format: 'YYYY-MM-DD HH:mm:ss:ms' }),
                  utilities.format.nestLike('가치마인드', {
                      prettyPrint: true,
                  }),
              ),
```

jn33-dev commented on project/src/middlewares/logger.service.ts:
```diff
@@ -0,0 +1,34 @@
+        new winstonDaily(dailyOptions('info')),
+        new winstonDaily(dailyOptions('warn')),
+        new winstonDaily(dailyOptions('error')),
+    ],
```
파일 저장 함수는 아래와 같이 maxFiles 옵션과 함께 패싱해주세요.
```
        new winstonDaily(dailyOptions('info', 7)),
        new winstonDaily(dailyOptions('warn', 14)),
        new winstonDaily(dailyOptions('error', 30)),
```

jn33-dev reviewed (changes requested):
세현님, 원래 있던 내장 로거는 어차피 작동을 안해서 삭제하시면 됩니다. 
그런데, 지금 로거 폴더를 삭제하고 다시 돌려보시면 아마 폴더가 생성이 안되는 것 같습니다. 
이 부분 제 컴퓨터 문제인지 한번 확인해주세요.

jn33-dev committed:
REFACTOR : 윈스턴 로거 옵션 수정 #43

Changes to be committed:
	modified:   src/app.module.ts
	renamed:    src/middlewares/logger.middleware.ts -> src/logger/logger.middleware.ts
	new file:   src/logger/winston.util.ts
	modified:   src/main.ts
	deleted:    src/middlewares/logger.service.ts

jn33-dev committed:
Merge branch 'dev' of https://github.com/gachimind/backend into feat/11

Conflicts:
	project/dist/tsconfig.build.tsbuildinfo

Changes to be committed:
	modified:   ../.github/workflows/main.yml
	modified:   src/games/dto/socketId-map.request.dto.ts
	modified:   src/games/entities/socketIdMap.entity.ts
	modified:   src/games/games.gateway.ts
	modified:   src/games/players.service.ts
	modified:   src/games/util/today.date.constructor.ts
	new file:   src/users/dto/token.map.request.dto.ts
	modified:   src/users/dto/update-user.dto.ts
	modified:   src/users/entities/token-map.entity.ts
	modified:   src/users/users.controller.ts
	modified:   src/users/users.service.ts

jn33-dev commented:
세현님, 윈스턴 로거 옵션 수정해서 올렸습니다.

지금 커멘트 달아 둔 것 중에 아직 해결 안 된 것 반영해서 마무리 부탁드립니다!

bsh714-dev committed:
FIX: NOTION API

bsh714-dev committed:
FIX : 로거파일 저장위치 변경, 그외 수정

jn33-dev commented on project/src/admin/admin.controller.ts:
```diff
@@ -7,7 +7,7 @@ export class AdminController {
     @Post('report')
-    async bugReport(@Body() Body: BugDto) {
+    async bugReport(@Body() body: BugDto) {
         if (!Body) {
```
위에 변수 바뀌면 블럭 스콥 안에도 다 바꿔주셔야 해요 세현님ㅠㅠ

bsh714-dev committed:
FIX : 로거, 노션 API

bsh714-dev committed:
FIX : 로거, 노션 API 수정

jn33-dev reviewed (approved):
고생 많으셨습니다. 세현님! 머지 해주세요!
